vmtest/download.py

```diff
@@ -286,7 +288,12 @@ def _download_thread(
 def download_in_thread(
     download_dir: Path, downloads: Iterable[Download]
 ) -> Generator[Iterator[Downloaded], None, None]:
-    q: "queue.Queue[Union[Downloaded, Exception]]" = queue.Queue()
+    # Downloading too many files before they can be used for testing runs the
+    # risk of filling up the limited disk space is Github Actions. Set a limit
+    # of no more than 5 files which can be downloaded ahead of time. This is a
+    # magic number with no testing, but should work.
+    maxsize = 5 if IN_GITHUB_ACTIONS else 0
```

Comment:
This would be cleaner as an argument passed to `download_in_thread()`. Other than that, this is a really clever solution.

Reply:
Thanks! Would you like `vmtest/__main__.py` to pass this parameter based on `GITHUB_ACTIONS` environment variable? There are two callers, but of course `setup.py` is the only one used in CI. But in case we switch over to `vmtest/__main__.py` we may want the logic in both places.

Reply:
Good catch, yeah, just copy the logic in both places. I have a dream of getting off setuptools entirely, and `vmtest/__main__.py` will replace this if that ever happens.